source/templates.tex

```diff
@@ -6572,10 +6572,9 @@
 shall appertain to an explicit instantiation.
 
 \pnum
-If the explicit instantiation is for a class or member class, the
-\grammarterm{elaborated-type-specifier} in the \grammarterm{declaration}
-shall include a \grammarterm{simple-template-id};
-otherwise, the \grammarterm{declaration}
+If the explicit instantiation is for a class or member class, the \grammarterm{declaration}
+shall consist of a single \grammarterm{elaborated-type-specifier} that includes a 
+\grammarterm{simple-template-id}; otherwise, the \grammarterm{declaration}
```

Comment:
I think the "shall consist of a single" part of this rule belongs in [temp.pre]p5:

> When such a declaration is used to declare a class template <ins>or class template specialization</ins>, no declarator is permitted.

As written, this doesn't seem quite right: requiring the *declaration* to consist of only an *elaborated-type-specifier* would disallow attributes.

But there's something really fishy about the rules here in general: we're starting (in both [temp.pre] and here) by assuming we know whether we're explicitly instantiating a class or something else, and there's no rules to say when we are doing so.

What we *really* mean is something like: if there is no *declarator*, then the template, template specialization, or explicit instantiation declaration is declaring a class or class template, and the decl-specifier-seq shall contain a single defining-type-specifier that is either a class-specifier, an enum-specifier, or an elaborated-type-specifier. And for an explicit instantiation, it shall be an elaborated-type-specifier that either includes a simple-template-id or whose nested-name-specifier includes a simple-template-id.